### PR TITLE
Allow admins and authors to see fastest solves when not registered

### PR DESCRIPTION
**Pull request by digit01Wave**
Allow admins and authors to see fastest solves even when they are not registered for the event

### CODE REVIEW

File: ServerCore/Pages/Events/FastestSolves.cshtml.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public async Task OnGetAsync(TeamPuzzleSortOrder? teamPuzzleSort, SinglePlayerPu
         private async Task updateTeamPuzzleStats()
         {
             bool isRegisteredUser = await this.IsRegisteredUser();
-            if (!isRegisteredUser)
+            if (EventRole == EventRole.play && !isRegisteredUser)
```
Comment (morganbr) [inline ServerCore/Pages/Events/FastestSolves.cshtml.cs:55]: For authors, this would need to filter the puzzles to the author's puzzles

Comment (digit01Wave) [reply ServerCore/Pages/Events/FastestSolves.cshtml.cs:55]: Shouldn't authors see the fastest solves even if they didn't write the puzzle?
After all, players can see the fastest solves of all puzzles

Comment (tabascq) [reply ServerCore/Pages/Events/FastestSolves.cshtml.cs:55]: As a player, the list of puzzles shown is filtered to what you have unlocked, so I agree with Morgan.

Comment (digit01Wave) [reply ServerCore/Pages/Events/FastestSolves.cshtml.cs:55]: Is there a reason we don't want authors to see the fastest solves page of all the puzzles?

Comment (tabascq) [reply ServerCore/Pages/Events/FastestSolves.cshtml.cs:55]: Exactly

Comment (digit01Wave) [reply ServerCore/Pages/Events/FastestSolves.cshtml.cs:56]: Makes sense. I updated it

**Review by tabascq — APPROVED** [review]

**Pull request merged**